### PR TITLE
Add find / wait for elements methods

### PR DESCRIPTION
**Pull request by atroche**
Still unsure about the whole `wait` vs `find` part of the API, and also the necessity of the "custom timeout" version, and also the `found_via_selector` field, but those can wait for another time.

### CODE REVIEW

File: src/browser/tab/mod.rs [hunk 1]
```diff
@@ -185,17 +187,31 @@ impl<'a> Tab {
...
     }
 
+    pub fn wait_for_elements(&'a self, selector: &'a str) -> Result<Vec<Element<'a>>, Error> {
+        debug!("Waiting for element with selector: {}", selector);
+        util::Wait::with_timeout(Duration::from_secs(15))
+            .until(|| {
+                if let Ok(elements) = self.find_elements(selector) {
+                    if elements
+                        .iter()
+                        .all(|element| element.get_midpoint().is_ok())
```
Comment (lukaslueg) [inline src/browser/tab/mod.rs]: Why is this here?

**Comment by lukaslueg** [comment]
The `found_via_selector`-item seems superfluous and, afaics, adds an unnecessary lifetime restriction to `Element`. It probably should go some time.

**Comment by lukaslueg** [comment]
The whole `find`/`wait` thing boils down to the fact that this is a synchronous API and waiting is the only thing one can do. There is no way to fundamentally solve concurrent events in a synchronous API. 

I'd suggest to simply remove all `find` variants and only use `wait`.

File: src/browser/tab/mod.rs (hunk 1 above)
Comment (atroche) [reply src/browser/tab/mod.rs]: It really shouldn't be! Removed from here and from `wait_for_element`.
